metrics/metrics_collector.go

```diff
@@ -0,0 +1,160 @@
+package metrics
+
+import (
+	"fmt"
+	"math"
+	"math/rand"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type metricCollector struct {
+	counters   []prometheus.Counter
+	gauges     []prometheus.Gauge
+	histograms []prometheus.Histogram
+	summarys   []prometheus.Summary
+	interval   int
+}
+
+var (
+	promRegistry = prometheus.NewRegistry() // local Registry so we don't get Go metrics, etc.
+)
+
+func newMetricCollector() metricCollector {
+	return metricCollector{}
+}
+
+func (mc *metricCollector) updateCounter() {
+	for {
+		time.Sleep(time.Second * time.Duration(mc.interval))
+		for idx := 0; idx < len(mc.counters); idx++ {
+			mc.counters[idx].Add(rand.Float64())
```

Comment:
```suggestion
		for _, c := range mc.counters {
			c.Add(rand.Float64())
```

Is there a reason for not ranging over the collection here?  It's more idiomatic to use this loop form where possible.